include/swift/AST/TypeRepr.h

```diff
@@ -1145,10 +1145,11 @@ class CompileTimeConstTypeRepr : public SpecifierTypeRepr {
 class FixedTypeRepr : public TypeRepr {
   Type Ty;
   SourceLoc Loc;
+  bool IsSelf;
```

Comment:
Why do we need this flag is this could be established based on `Ty`?

Reply:
AFAICT the type being written as Self isn't inspectable anywhere, but in the case of the index that impacts whether or not the reference should be implicit or explicit

Reply:
I wonder whether it would make sense to have a special Repr node for this instead of just a flag?

Reply:
If this is the only place where we care about the difference with this maybe not? I guess it would either be this or a subclass of FixedTypeRepr that could be used to determine if they different?

Reply:
IMO a new TypeRepr would be best, `Self` is the only user of `FixedTypeRepr` outside of `TypeExpr::createImplicitHack`, and that really ought to be the only user.